src/software/ai/navigator/path_planner/hrvo/hrvo_agent.cpp

```diff
@@ -87,6 +95,8 @@ std::vector<RobotId> HRVOAgent::computeNeighbors(
     double dist_to_obstacle_threshold_squared =
         std::min(std::pow(MAX_NEIGHBOR_SEARCH_DIST, 2),
                  (position - current_destination).lengthSquared());
+    dist_to_obstacle_threshold_squared = std::max(std::pow(ROBOT_MAX_RADIUS_METERS, 2),
+                                                  dist_to_obstacle_threshold_squared);
```

Comment:
I've slightly changed the values for this logic and added docs, but its basically added so adjacent robots don't push each other. 
An example of this would be when you have two robots, with robot 1 (e.g. a crease defender) being at the destination and robot 2 has its destination within two robot radii of robot 1. With the previous logic, since robot 2 is closer to the destination than it is to robot 1, it would ignore robot 1 and try to push it away (not knowing it's there). Likewise, robot 1 would push robot 2 back, also not knowing it's there. This updated neighbour distance threshold makes it such that the robots see adjacent robots and stops them from pushing each other.
This may have some side effects, so we would need to see how it behaves in other scenarios as well (e.g. what happens if an enemy robots comes close, does it move away letting it pass?!).